ffi/framebuffer_linux.lua

```diff
@@ -245,11 +269,47 @@ function framebuffer:init()
     end
 
     self.bb:fill(BB.COLOR_WHITE)
+end
 
-    framebuffer.parent.init(self)
+-- If the device can stomach it (this feature is frontend opt-in) perform hardware rotations
+-- FIXME: What about viewports? Do we need to revert back to SW rotation mode?
+function framebuffer:setRotationMode(mode)
+    if not self.device.usingForcedRotation then
+        -- Go the standard route
+        return self.parent.setRotationMode(self, mode)
+    end
+    self.debug("setRotationMode:", mode, "old:",self.cur_rotation_mode)
+    if mode ~= self.cur_rotation_mode then
+        -- Do this dance only if rotation actually changes
+        self.cur_rotation_mode = mode
+        self:forceRotation()
+        self:reinit()
+        self:restoreRotation()
```

Comment:
Needing a restore there is extremely PB-specific, that's basically the only thing mildy bothering me with this implementation ;).

Reply:
And of course I'd missed the `needsRestoredRotation` device cap, so, never mind me ;).

Reply:
There's still device specifity though: We do forceRotation on every batch of paint. This is not necessarily needed for devies that can set and "hold" the rotate once it set once in here. But then again, better safe than sorry, and the call isn't all that expensive compared to bunch of paints, ergo, no bothering with special cap just to signal this.

Reply:
Mhhh, I spent a few minutes trying to understand what this is all about ... and didn't :)
Could you add a few comments about what all this new is about, why you forceRotation to soon after restoreRotation please...?

Reply:
Thanks for the added comments, that's clearer and I get it. May be just add a comment mentionning "i.e. on Pocketbooks, where xxx/yyy may draw over koreader" so we don't have to wonder who would do that on Kobo or Kindle :)